nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java

```diff
@@ -82,39 +95,51 @@ public class PutMongo extends AbstractMongoProcessor {
                 + "otherwise it is ignored")
         .required(true)
+        .dependsOn(MODE, MODE_UPDATE)
         .allowableValues("true", "false")
         .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
         .defaultValue("false")
         .build();
     static final PropertyDescriptor UPDATE_QUERY_KEY = new PropertyDescriptor.Builder()
         .name("Update Query Key")
-        .description("Key name used to build the update query criteria; this property is valid only when using update mode, "
-                + "otherwise it is ignored. Example: _id")
+        .description("Comma separated key names used to build the update query criteria. Their values are taken from incoming flowfile. Example: _id")
         .required(false)
+        .dependsOn(MODE, MODE_UPDATE)
         .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
         .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
         .build();
     static final PropertyDescriptor UPDATE_QUERY = new PropertyDescriptor.Builder()
         .name("putmongo-update-query")
         .displayName("Update Query")
-        .description("Specify a full MongoDB query to be used for the lookup query to do an update/upsert.")
+        .description("Specify a full MongoDB query to be used for the lookup query to do an update/upsert. NOTE: this field is ignored if the '%s' value is not empty."
+            .formatted(UPDATE_QUERY_KEY.getDisplayName()))
         .required(false)
+        .dependsOn(MODE, MODE_UPDATE)
         .addValidator(JsonValidator.INSTANCE)
         .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
         .build();
 
-    static final PropertyDescriptor UPDATE_MODE = new PropertyDescriptor.Builder()
+    static final PropertyDescriptor UPDATE_OPERATION_MODE = new PropertyDescriptor.Builder()
         .displayName("Update Mode")
         .name("put-mongo-update-mode")
         .required(true)
+        .dependsOn(MODE, MODE_UPDATE)
         .allowableValues(UPDATE_WITH_DOC, UPDATE_WITH_OPERATORS)
-        .defaultValue(UPDATE_WITH_DOC.getValue())
+        .defaultValue(UPDATE_WITH_DOC)
         .description("Choose an update mode. You can either supply a JSON document to use as a direct replacement " +
                 "or specify a document that contains update operators like $set, $unset, and $inc. " +
                 "When Operators mode is enabled, the flowfile content is expected to be the operator part " +
                 "for example: {$set:{\"key\": \"value\"},$inc:{\"count\":1234}} and the update query will come " +
                 "from the configured Update Query property.")
          .build();
+    static final PropertyDescriptor MONGO_UPDATE_MODE = new PropertyDescriptor.Builder()
+        .name("Mongo Update Query Mode")
+        .displayName("Mongo Update Query Mode")
```

Comment:
Although it is difficult to differentiate this property from the existing `Update Mode` property, I think another name would be better. One option could be `Update Method`, another option could be `Update Strategy`. Although it is still similar to `Update Mode`, it would provide some differentiation with using the word `Mongo` and `Mode` unnecessarily. Also, the `displayName()` can be omitted when it matches the property name.

Depending on the name chosen, I also recommend updating the variable references to use `updateMethod` or `updateStrategy` instead of `updateQueryMode`.

```suggestion
    static final PropertyDescriptor UPDATE_METHOD = new PropertyDescriptor.Builder()
        .name("Update Method")
```